express-delete/main.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express');
+
+const app = express();
+
+const grades = [
+  {
+    id: 1,
+    name: 'Beeg Yoshi',
+    course: 'Meming',
+    grade: 9001
+  },
+  {
+    id: 2,
+    name: 'Kirby',
+    course: 'Inhale',
+    grade: 7000
+  }
+];
+
+// app.use((req, res) => {
+//   console.log(req.param);
+// });
+
+app.get('/api/grades', (req, res) => {
+  res.json(grades);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  grades.splice([req.param.id - 1], 1);
```

Comment:
The `id` of a grade is not always `1` different from its index in the array. Avoid doing math on ids to calculate indexes, they're not actually related. Instead use a loop.

Reply:
revised!